app/routes/surveys/components/Question.js

```diff
@@ -0,0 +1,237 @@
+import Option from './Option';
+import { TextInput, TextArea, CheckBox } from 'app/components/Form';
+import Dropdown from 'app/components/Dropdown';
+import styles from './surveys.css';
+import Icon from 'app/components/Icon';
+import { Link } from 'react-router';
+import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
+
+type Props = {
+  updateQuestion: Object => void,
+  deleteQuestion: number => Promise<*>,
+  question: Object,
+  nr: number
+};
+
+type State = {
+  shouldUpdate: boolean,
+  options: Array<Object>,
+  pickerOpen: boolean
+};
+
+type QuestionTypeSelectorProps = {
+  questionType: number
+};
+
+function QuestionTypeSelector({ questionType }: QuestionTypeSelectorProps) {
+  const questionTypeIntToString = {
+    '1': 'Multiple Choice',
+    '2': 'Sjekkboks',
+    '3': 'Fritekst'
+  };
+  const questionTypeIntToIcon = {
+    '1': 'radio-button-on',
+    '2': 'checkbox',
+    '3': 'more'
+  };
+  return (
+    <Link className={styles.questionTypeSelector}>
+      <span style={{ width: '170px', display: 'inline-block' }}>
+        <Icon
+          name={questionTypeIntToIcon[String(questionType)]}
+          style={{ marginRight: '15px' }}
+        />
+        {questionTypeIntToString[String(questionType)]}
+      </span>
+      <Icon name="arrow-down" size={24} className={styles.typeIcon} />
+    </Link>
+  );
+}
+
+type DropdownItemsProps = {
+  setQuestionType: number => void
+};
+
+function DropdownItems({ setQuestionType }: DropdownItemsProps) {
+  return (
+    <Dropdown.List>
+      <Dropdown.ListItem>
+        <Link onClick={() => setQuestionType(1)}>
+          <strong>Multiple Choice</strong>
+          <Icon name="radio-button-on" size={24} />
+        </Link>
+      </Dropdown.ListItem>
+      <Dropdown.ListItem>
+        <Link onClick={() => setQuestionType(2)}>
+          Sjekkboks
+          <Icon name="checkbox" size={24} />
+        </Link>
+      </Dropdown.ListItem>
+      <Dropdown.ListItem>
+        <Link onClick={() => setQuestionType(3)}>
+          Fritekst
+          <Icon name="more" size={24} />
+        </Link>
+      </Dropdown.ListItem>
+    </Dropdown.List>
+  );
+}
+
+class Question extends Component<Props, State> {
+  state = {
+    shouldUpdate: false,
+    options: [{ nr: 0, optionText: '' }],
+    pickerOpen: false
+  };
+
+  componentDidMount() {
+    const options = this.props.question.options
+      .map((option, i) => ({ ...option, nr: i }))
+      .concat({
+        nr: this.props.question.options.length,
+        optionText: ''
+      });
+    this.setState({ options });
+  }
+
+  componentDidUpdate() {
+    if (this.state.shouldUpdate) {
+      this.props.updateQuestion({
+        ...this.props.question,
+        options: this.state.options
+      });
+      this.setState({ shouldUpdate: false });
+    }
+  }
+
+  updateOptions = (option: Object) => {
+    const options = this.state.options.slice();
+
+    const oldOption = options.find(o => o.nr === option.nr);
```

Comment:
isn't `nr` the index? do you need a `find`?

Reply:
As explained above, I thought of `nr` as an ID, meaning an option would keep its `nr` even if previous elements in the `options` array were deleted.